packages/database/src/sync/resolveDuplicatedPatientDisplayIds.ts

```diff
@@ -20,8 +20,11 @@ export const resolveDuplicatedPatientDisplayIds = async (
   changes: SyncSnapshotAttributes[],
 ): Promise<SyncHookSnapshotChanges | undefined> => {
   const patientDisplayIds = changes.filter((c) => !c.isDeleted).map((c) => c.data.displayId);
+  const patientIds = changes.filter((c) => !c.isDeleted).map((c) => c.data.id);
+
+  // Find existing patients that have the same displayIDs and are not the same with the incoming patients
   const existingPatientsWithDuplicatedDisplayIds = await PatientModel.findAll({
-    where: { displayId: { [Op.in]: patientDisplayIds } },
+    where: { displayId: { [Op.in]: patientDisplayIds }, id: { [Op.notIn]: patientIds } },
```

Comment:
When fetching existing patients with duplicated display IDs, it should exclude the current patients themselves